### PR TITLE
Revert "Remove concurrency and cleanup workflows"

### PR DESCRIPTION
**Pull request by rbren**
Reverts All-Hands-AI/OpenHands#3524

### CODE REVIEW

**Comment by rbren** [comment]
FYI @mamoodi--this broke the multi-arch builds. The `create_manifest` step pulls the ARM and AMD images into a single tag

**Comment by rbren** [comment]
Actually I'm not 100% sure that this caused the issue--we'll see how the docker build goes here

File: .github/workflows/ghcr_app.yml [hunk 1]
```diff
@@ -78,12 +79,12 @@ jobs:
           path: /tmp/${{ matrix.image }}_${{ steps.capture-last-tag.outputs.last_tag }}_${{ matrix.platform }}.tar
           retention-days: 14
 
-  # Push the OpenHands Docker images to the ghcr.io repository
+  # Push the OpenHands and sandbox Docker images to the ghcr.io repository
```
Comment (xingyaoww) [inline .github/workflows/ghcr_app.yml:82]:
```suggestion
  # Push the OpenHands Docker images to the ghcr.io repository
```

File: .github/workflows/ghcr_runtime.yml [hunk 2]
```diff
@@ -247,7 +249,7 @@ jobs:
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
-  # Checks that all runtime tests have passed
+  # New job to indicate all runtime tests have passed
```
Comment (xingyaoww) [inline .github/workflows/ghcr_runtime.yml:252]:
```suggestion
  # Checks that all runtime tests have passed
```

**Pull request merged**